File: leds_unity/Assets/shooter/Explotion.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

namespace Shooter
{
    public class Explotion
    {
        public int ledId;
        public float width = 7;
        public Color color;
        public Color originalColor;
        int numLeds;
        public Shoots shoots;
        public float dir;
        float timer;
        public List<int> leds;
        public bool isOn;

        public void Init(Shoots shoots, int numLeds, int ledId, Color _color)
        {
            Debug.Log("Init " + numLeds);
            this.ledId = ledId;
            leds = new List<int>();
            for (int a = 0; a < width; a++)
                leds.Add(0);
            this.shoots = shoots;
            this.numLeds = numLeds;
            this.originalColor = _color;
            this.color = _color;
            Restart(ledId);

        }
        public void Restart(int ledId)
        {
            this.ledId = ledId;
            timer = 0;
            isOn = true;
            int thisLed = ledId - (int)(Mathf.Floor(width/2));
            if (thisLed < 0) thisLed = numLeds-1 + thisLed;
            else if (thisLed >= numLeds) thisLed = thisLed - numLeds;
            for (int a = 0; a < width; a++)
            {
                if (thisLed < 0) thisLed = numLeds + thisLed;
                else if (thisLed >= numLeds) thisLed = thisLed - numLeds;
                leds[a] = thisLed;
                thisLed++;
            }
        }
        public void OnUpdate(float deltaTime)
        {
            timer += deltaTime;
            if (timer > 1)
                isOn = false;
        }
       
    }

}

File: micropython/bubbles/explotion.py
import urandom

class ExplotionParticle:
    def Init(self, numLeds, ledId, dir, color):
        self.ledId = ledId
        self.numLeds = numLeds
        self.Restart(ledId, dir, color)

    def Restart(self, ledId, dir, color):        
        self.speed = urandom.uniform(20, 220)
        self.color = 10
        self.oColor = color
        self.timer = 0
        self.isOn = True
        self.dir = dir
        self.ledId = ledId
        self.pos = ledId
        self.alpha = 0.9

    def OnUpdate(self, deltaTime):
        self.timer += deltaTime
        if self.timer > 0.25:
            self.color = self.oColor
        self.speed /= 1.1
        self.pos += deltaTime * self.speed * self.dir
        self.alpha -= deltaTime / (self.speed/50)
        if self.alpha < 0:
            self.isOn = False
            self.alpha = 0
            return

        self.ledId = int(self.pos)
        if self.ledId < 0:
            self.ledId = self.numLeds - 1 + self.ledId
        elif self.ledId >= self.numLeds:
            self.ledId = self.ledId - self.numLeds

        if self.timer > 1:
            self.isOn = False
